lua/lean/widgets.lua

```diff
@@ -37,20 +34,33 @@ local BufRenderer = {
 }
 BufRenderer.__index = BufRenderer
 
+---@class ElementNewArgs
+---@field events table<string, fun()> @event function map
+---@field text string @the text to show when rendering this element
+---@field name string @a named handle for this element, used when path-searching
+---@field hlgroup? string|fun():string @the highlight group for this element's text, or a function that returns it
+---@field highlightable boolean @(for buffer rendering) whether to highlight this element when hovering over it
+---@field children Element[] @this element's children
+
 ---Create a new Element.
+---@param args ElementNewArgs
 ---@return Element
-function Element:new(obj)
-  obj = obj or {}
-  local children = obj.children or {}
-  obj.children = nil
-  return setmetatable(
-    vim.tbl_extend("keep", obj, { events = {}, __children = children }),
```

Comment:
`tbl_extend` was 12% of the runtime btw.

Reply:
Eesh.